doc/network.md

```diff
@@ -89,6 +89,54 @@ Size: 0 if non-present, else size of `X`.
 
 ## Packets
 
+### Game-related Packets
+
+#### `plasmid:game/player_add`
+
+Packet sent to all players of a game space when a player is joined to a game space.
+The player that was joined to the game space also receives this packet.
+
+Direction: `C<-S`
+
+| Fields         | Type       | Description                                                     |
+|:--------------:|:----------:|:----------------------------------------------------------------|
+| game_type_id   | Identifier | The identifier of the game type.                                |
+| game_type_name | Text       | The name of the game type.                                      |
+| game_id        | Identifier | The identifier of the game.                                     |
```

Comment:
On second thought, I'm not sure if this should be included in the packet. Relying on game ids is never a good idea.

Reply:
Is there any reason why packet consumers should not be able to have access to the game ID?

Reply:
They can’t and shouldn’t be relied on by anything. Games don’t have an ID- hence the rename to source. It’s just where the game was loaded from, and that might not even be present in the case of an anonymously created game.

Reply:
Packet consumers would be required to handle anonymous games, then. If the identifier was removed, packet consumers would just end up relying on text, which is worse. After all, games can have no text as well (though, this just falls back rather than being `null`).

Reply:
They should probably rather be passed all the game custom values.

Reply:
Most games don't pass any custom values, and if they did it would be at the mercy of the game's developers, similar to the name.

Reply:
Hardcoding per JSON file is arguably worse. It can change, and new JSONs can be added and then the handling needs to be updated. Nothing should rely on the JSON file that creates the config- the config itself must provide enough information in itself. I think it's fine if the consumer must rely on the game providing the information it needs.

Reply:
Packet consumers won't be stopped from hardcoding things; it'll just make packets harder to consume. I think the best solution is to document that it is generally a bad idea to depend on game IDs for longer-term cases.

Reply:
Giving the option to use game sources will encourage it, where by design consumer should instead make use of custom data values to be provided by games.

Reply:
Custom data values would be useless or non-existent in most cases. I see the game ID as the only solution to certain use cases, where anonymous games can be handled as well just fine.

Reply:
Can you elaborate on these cases?

Reply:
For example, a tracker client mod would store the games in a map like the following:

```json
{
    "bedwars:the_end_bed_wars": 1,
    "bedwars:cubes_bed_wars": 1
}
```

Anonymous games would never appear in this map due to a lack of a game ID.

Reply:
Why would it store things like this as opposed to having some value in the game's config?

Reply:
Would that custom value not just be the same as the game ID? In addition, would all anonymous games have that custom value as well? It would duplicate the existing game ID for not much benefit.

Reply:
No- it would provide whatever values are needed directly. Anonymous games are an example of where things would not have a game id, but really the motivation in this case is they can't be considered to be stable. A new game config can be added, or a config can be moved around- but then you have to update the map within the tracker to consider that.

Reply:
The game ID is the thing that this map would need, and for the problem the tracker aims to solve, there is no way around it; any attempts at providing a custom value replacement would just duplicate the ID.

I think packet consumers can handle the task of recognizing that the given game IDs are not stable and may not exist. The tracker would ignore games without IDs and treat games with changed IDs as entirely separate games, unless the user chooses to merge two game IDs.

Reply:
The point is to not have the map in the first place, but rather the values within the map provided within the game. What do you imagine this map would be used to store?

Reply:
The map would store play counts for each game.